### PR TITLE
Update expression and anomaly detector documentation

### PR DESCRIPTION
**Pull request by kkondaka**
### Description
Data Prepper supports arithmetic operators but the documentation is not updated to reflect that.
Also, Random Cut forest in anomaly detector processor support a new option which is not reflected in the documentation.

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._

### Version
_List the OpenSearch version to which this PR applies, e.g. 2.14, 2.12--2.14, or all._

### Frontend features
_If you're submitting documentation for an OpenSearch Dashboards feature, add a video that shows how a user will interact with the UI step by step. A voiceover is optional._ 

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _data-prepper/pipelines/expression-syntax.md
Comment (dlvenable) [inline _data-prepper/pipelines/expression-syntax.md:33]:
```suggestion
| `*`, `/`             | Multiplication and division operators                 | Left to right |
```

Division is not a proper noun and need not be capitalized.

File: _data-prepper/pipelines/expression-syntax.md
Comment (dlvenable) [inline _data-prepper/pipelines/expression-syntax.md:34]:
```suggestion
| `+`, `-`             | Addition and subtraction operators                    | Left to right |
```

Substraction is not a proper noun and need not be capitalized.

File: _data-prepper/pipelines/expression-syntax.md
Comment (dlvenable) [inline _data-prepper/pipelines/expression-syntax.md:230]:
```suggestion
| `+`                  | String concatenation operator   | No                   | `/status_code + /message + "suffix"`
```

File: _data-prepper/pipelines/expression-syntax.md
Comment (dlvenable) [inline _data-prepper/pipelines/expression-syntax.md:231]:
```suggestion
| `+`, `-`             | Arithmetic addition and subtraction operators | No      | `/status_code + length(/message) - 2`
```

File: _data-prepper/pipelines/expression-syntax.md
Comment (dlvenable) [inline _data-prepper/pipelines/expression-syntax.md:232]:
```suggestion
| `*`, `/`             | Multiplication and division operators | No              | `/status_code * length(/message) / 3`
```

**Review by dlvenable — DISMISSED** [review]

File: _data-prepper/pipelines/configuration/processors/anomaly-detector.md
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/anomaly-detector.md:56]:
```suggestion
| `output_after` | 32 | N/A | Specifies the number of events to process before outputting any detected anomalies. |
```

File: _data-prepper/pipelines/expression-syntax.md
Comment (vagimeli) [inline _data-prepper/pipelines/expression-syntax.md:107]:
```suggestion
The following are example arithmetic expressions: 
```

File: _data-prepper/pipelines/expression-syntax.md
Comment (vagimeli) [inline _data-prepper/pipelines/expression-syntax.md:117]:
```suggestion
The following are example arithmetic expressions that can be used in conditional expressions: 
```

File: _data-prepper/pipelines/expression-syntax.md
Comment (vagimeli) [inline _data-prepper/pipelines/expression-syntax.md:98]:
```suggestion
Arithmetic expressions enable basic mathematical operations like addition, subtraction, multiplication, and division. These expressions can be combined with conditional expressions to create more complex conditional statements. The available arithmetic operators are +, -, *, and /. The syntax for using the arithmetic operators is as follows:
```

File: _data-prepper/pipelines/expression-syntax.md
Comment (vagimeli) [inline _data-prepper/pipelines/expression-syntax.md:134]:
```suggestion
The following are example string concatenation expressions:
```

File: _data-prepper/pipelines/expression-syntax.md
Comment (vagimeli) [inline _data-prepper/pipelines/expression-syntax.md:143]:
```suggestion
The following are example string concatenation expressions that can be used in conditional expressions:
```

File: _data-prepper/pipelines/expression-syntax.md
Comment (vagimeli) [inline _data-prepper/pipelines/expression-syntax.md:128]:
```suggestion
String concatenation expressions enable you to combine strings to create new strings. These concatenated strings can also be used within conditional expressions. The syntax for using string concatenation is as follows:
```

File: _data-prepper/pipelines/expression-syntax.md
Comment (vagimeli) [inline _data-prepper/pipelines/expression-syntax.md:153]:
```suggestion
Certain symbols, such as ^, %, xor, =, +=, -=, *=, /=, %=, ++, --, and ${<text>}, are reserved for potential future functionality or extensions, even though they are not currently used in the expression syntax. Reserved symbols include `^`, `%`, `xor`, `=`, `+=`, `-=`, `*=`, `/=`, `%=`, `++`, `--`, and `${<text>}`.
```

**Review by vagimeli — APPROVED** [review]
Please see the writing edits for clarity and conciseness. Once these changes are approved, I'll move this PR into editorial and then merge it.

**Comment by vagimeli** [comment]
@dlvenable I've completed the doc review and accepted your edits and comments. I also edited text for readability and style. Please review and approve at your availability, and then I'll move this forward for publishing. Thank you.

**Comment by vagimeli** [comment]
> @dlvenable I've completed the doc review and accepted your edits and comments. I also edited text for readability and style. Please review and approve at your availability, and then I'll move this forward for publishing. Thank you.

@kkondaka @vagimeli Doc review is complete and ready for your approval. Thanks!

File: _data-prepper/pipelines/configuration/processors/anomaly-detector.md
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/anomaly-detector.md:56]:
```suggestion
| `output_after` | 32 | N/A | Indicates the number of events to consume before outputting anomalies. |
```

**Comment by vagimeli** [comment]
> > @dlvenable I've completed the doc review and accepted your edits and comments. I also edited text for readability and style. Please review and approve at your availability, and then I'll move this forward for publishing. Thank you.
> 
> @kkondaka @vagimeli Doc review is complete and ready for your approval. Thanks!

@kkondaka Please provide a status update for this PR. It's awaiting your acceptance of my doc review changes and @dlvenable approval. Thank you.

**Comment by vagimeli** [comment]
> > > @dlvenable I've completed the doc review and accepted your edits and comments. I also edited text for readability and style. Please review and approve at your availability, and then I'll move this forward for publishing. Thank you.
> > 
> > 
> > @kkondaka @vagimeli Doc review is complete and ready for your approval. Thanks!
> 
> @kkondaka Please provide a status update for this PR. It's awaiting your acceptance of my doc review changes and @dlvenable approval. Thank you.

@kkondaka @dlvenable Please provide a status update for this PR. It's awaiting your acceptance of my doc review changes and @dlvenable approval. Thank you.

**Comment by Naarcha-AWS** [comment]
Reviewed this once again, looks like the original tech review by David was implemented. Dismissed his review and added Melissa's doc review. Merging now.

**Pull request merged**
